### PR TITLE
0.31.0, then 0.33.0, and add Deregister

### PR DESCRIPTION
**Pull request by digitalextremist**
This is a round-up of three sets of changes accumulating in my fork:

- [x] Make `0.31.0` compatible ( instead of #15 )
- [x] Add `deregister` support to `ZMQ::Poller` per #16 
- [x] Make `0.33.0` compatible per #18

### CODE REVIEW

File: src/zeromq/poller.cr [hunk 1]
```diff
@@ -59,4 +59,16 @@ class ZMQ::Poller
   def register_writable(socket)
     register(socket, POLLOUT)
   end
+
+  def deregister( socket : Socket, type = POLLIN | POLLOUT)
```
Comment (veelenga) [inline src/zeromq/poller.cr]:
```suggestion
  def deregister(socket : Socket, type = POLLIN | POLLOUT)
```

File: src/zeromq/socket.cr [hunk 2]
```diff
@@ -304,12 +305,15 @@ module ZMQ
...
-      @read_event = nil
-      @write_event.try &.free
-      @write_event = nil
+      @read_event.consume_each &.free
+      @write_event.consume_each &.free
       @closed = true
       LibZMQ.close @socket
     end
+
+    #de Copied from ::IO itself, since IO::Evented does not have this:
```
Comment (veelenga) [inline src/zeromq/socket.cr]:
```suggestion
    # Copied from ::IO itself, since IO::Evented does not have this:
```

**Review by veelenga — COMMENTED** [review]
Looks good. Is there a way to fix CI ?

**Comment by smalls12** [comment]
I fixed the travis build on on this pull request, thinking it would just append here ( it did not ).
I can pull request into master if that's better?
Note, the fix only made `crystal spec` run, it currently fails + times out.

**Comment by veelenga** [comment]
@smalls12 be free to open a pull request to fix a build in this repo. Thanks in advance

**Pull request merged**